tests/unit/amo/components/TestUserProfileEdit.js

```diff
@@ -0,0 +1,329 @@
+
+    expect(root.find('.UserProfileEdit-username')).toHaveLength(1);
+  });
+
+  it('renders a disabled username input field when not ready', () => {
+    const root = renderUserProfileEdit({ params: { username: 'not-ready' } });
+
+    expect(root.find('.UserProfileEdit-username')).toHaveProp('disabled', true);
+  });
+
+  it('renders a disabled "email" input field', () => {
+    const root = renderUserProfileEdit({ params: { username: 'tofumatt' } });
+
+    expect(root.find('.UserProfileEdit-email')).toHaveLength(1);
+    expect(root.find('.UserProfileEdit-email')).toHaveProp('disabled', true);
+  });
+
+  it('renders a help text for the email and homepage fields', () => {
+    const root = renderUserProfileEdit({ params: { username: 'tofumatt' } });
+
+    expect(root.find('.UserProfileEdit--help')).toHaveLength(2);
+  });
+
+  it('renders a displayName input field', () => {
+    const root = renderUserProfileEdit({ params: { username: 'tofumatt' } });
+
+    expect(root.find('.UserProfileEdit-displayName')).toHaveLength(1);
+  });
+
+  it('renders a disabled displayName input field when not ready', () => {
+    const root = renderUserProfileEdit({ params: { username: 'not-ready' } });
+
+    expect(root.find('.UserProfileEdit-displayName'))
+      .toHaveProp('disabled', true);
+  });
+
+  it('renders a homepage input field', () => {
+    const root = renderUserProfileEdit({ params: { username: 'tofumatt' } });
+
+    expect(root.find('.UserProfileEdit-homepage')).toHaveLength(1);
+  });
+
+  it('renders a disabled homepage input field when not ready', () => {
+    const root = renderUserProfileEdit({ params: { username: 'not-ready' } });
+
+    expect(root.find('.UserProfileEdit-homepage')).toHaveProp('disabled', true);
+  });
+
+  it('renders a location input field', () => {
+    const root = renderUserProfileEdit({ params: { username: 'tofumatt' } });
+
+    expect(root.find('.UserProfileEdit-location')).toHaveLength(1);
+  });
+
+  it('renders a disabled location input field when not ready', () => {
+    const root = renderUserProfileEdit({ params: { username: 'not-ready' } });
+
+    expect(root.find('.UserProfileEdit-location')).toHaveProp('disabled', true);
+  });
+
+  it('renders a occupation input field', () => {
+    const root = renderUserProfileEdit({ params: { username: 'tofumatt' } });
+
+    expect(root.find('.UserProfileEdit-occupation')).toHaveLength(1);
+  });
+
+  it('renders a disabled occupation input field when not ready', () => {
+    const root = renderUserProfileEdit({ params: { username: 'not-ready' } });
+
+    expect(root.find('.UserProfileEdit-occupation')).toHaveProp('disabled', true);
+  });
+
+  it('renders a biography input field', () => {
+    const root = renderUserProfileEdit({ params: { username: 'tofumatt' } });
+
+    expect(root.find('.UserProfileEdit-biography')).toHaveLength(1);
+  });
+
+  it('renders a disabled biography input field when not ready', () => {
```

Comment:
Is it really valuable to have all of these "it renders a disabled..." as separate tests? I think it might be acceptable to just combine them into a single test which checks all of the fields for disabled.

Reply:
We tend to write a test case for each intent.

Reply:
We could maybe merge some of them though.